### ISSUE TITLE
Undefined/upstream function calls cause return type errors

### ISSUE THREAD

**Issue by Rarst** [vimeo/psalm#1790]
I use a bunch of upstream functions, which are not present for the scan so I just suppress UndefinedFunction in config.

However I have noticed that whenever one is used in a method it causes  MixedInferredReturnType, even when the function itself _has no impact on return_. Just it being called at some point breaks return detection.

Couple examples.

This function always passes input through:

```php
/**
	 * Capture start time of HTTP request
	 *
	 * @param false|array|\WP_Error $false Whether to preempt an HTTP request's return value. Default false.
	 * @param array                 $args  HTTP request arguments.
	 * @param string                $url   The request URL.
	 *
	 * @return false|array|\WP_Error
	 */
public function pre_http_request( $false, array $args, string $url ) {
		$this->start( $url, 'http' );
		$this->callers[ $url ] = wp_debug_backtrace_summary( __CLASS__, 5 );
		return $false;
	}
```
https://github.com/Rarst/laps/blob/1e9a6000965579b2842d4c20c5a66a3f02fe14a9/src/Record/Collector/Http_Collector.php#L38-L53
```
ERROR: MixedInferredReturnType - src\Record\Collector\Http_Collector.php:45:13 - 
Could not verify return type 'array<array-key, mixed>|false|WP_Error' for Rarst\Laps\Record\Collector\Http_Collector::pre_http_request
         * @return false|array|\WP_Error
```

This one straight returns new instance:

```php
private function create_record( array $entry ): Record {
		return new Record(
			plugin_basename( $entry['name'] ),
			$entry['origin'],
			$entry['duration'],
			'',
			'plugin'
		);
	}
```
https://github.com/Rarst/laps/blob/1e9a6000965579b2842d4c20c5a66a3f02fe14a9/src/Record/Collector/Plugin_Load_Collector.php#L58-L66
```
ERROR: MixedInferredReturnType - src\Record\Collector\Plugin_Load_Collector.php:58:50 - 
Could not verify return type 'Rarst\Laps\Record\Record' for Rarst\Laps\Record\Collector\Plugin_Load_Collector::create_record
        private function create_record( array $entry ): Record {
```

**Comment by muglug** [comment]
Would you be able to reproduce the issue at https://psalm.dev?

**Comment by Rarst** [comment]
Here is one, based on return object example https://psalm.dev/r/415fdecf9c

**Comment by muglug** [comment]
At the very least this should be fine: https://psalm.dev/r/fc07d312e0

**Comment by muglug** [comment]
Fixed in 56e66af

**Comment by Rarst** [comment]
That fixed returns! And also unleashed a ton of errors, which were probably hidden by it... 😅
